packages/website/pages/upload.js

```diff
@@ -0,0 +1,131 @@
+import { getToken, API } from "../lib/api";
+import { useRouter } from "next/router";
+import { Web3Storage } from "web3.storage";
+import { useQueryClient } from "react-query";
+import { useDropzone } from "react-dropzone";
+import { useRef, useState } from "react";
+import Button from "../components/button.js";
+
+export function getStaticProps() {
+  return {
+    props: {
+      title: "Upload - Web3 Storage",
+      redirectTo: "/",
+      needsUser: true,
+    },
+  };
+}
+
+export default function Upload() {
+  const router = useRouter();
+  const queryClient = useQueryClient();
+  const [uploading, setUploading] = useState(false);
+  const [inputFile, setInputFile] = useState(null);
+  const inputRef = useRef();
+  const [percentComplete, setPercentComplete] = useState(0);
+
+  /**
+   * @param {import('react').ChangeEvent<HTMLFormElement>} e
+   */
+  async function handleUploadSubmit(e) {
+    e.preventDefault();
+    const data = new FormData(e.target);
+    const file = data.get("file");
+    if (file && file instanceof File) {
+      await uploadFile(file)
+    }
+  }
+
+  async function uploadFile(file) {
+    const client = new Web3Storage({
+      token: await getToken(),
+      endpoint: new URL(API),
+    });
+    setUploading(true);
+    try {
+      let totalBytesSent = 0;
+      await client.put([file], {
+        name: file.name,
+        onStoredChunk: (size) => {
+          totalBytesSent += size;
+          setPercentComplete(Math.round((totalBytesSent / file.size) * 100));
+        },
+      });
+    } finally {
+      await queryClient.invalidateQueries("get-uploads");
+      setUploading(false);
+      router.push("/files");
+    }
+  }
+
+  const onDrop = (acceptedFiles) => uploadFile(acceptedFiles[0]);
+  const { getRootProps, isDragActive } = useDropzone({ onDrop, multiple: false });
+
+  const openInput = () => inputRef.current.click()
+
+  return (
+    <main className="p-4 sm:px-16 mt-4 sm:mt-16 text-w3storage-purple">
+      <div className="mw9 pv3 ph3 ph5-ns min-vh-100 flex flex-col items-center">
```

Comment:
We can remove the tachyon classes in this file.

Reply:
@rafaelramalho19 can do? i assume they do nothing